src/widgets/index/index.ts

```diff
@@ -25,6 +25,82 @@ import {
   mergeSearchParameters,
 } from '../../lib/utils';
 
+import { WidgetType, WidgetParams, Schema } from '../../../schema';
+
+function Pbf() {
+  this.buf = new Uint8Array(0);
+  this.pos = 0;
+  this.type = 0;
+  this.length = 0;
+}
+
+Pbf.prototype = {
+  finish() {
+    this.length = this.pos;
+    this.pos = 0;
+    return this.buf.subarray(0, this.length);
+  },
+
+  writeVarint(val) {
+    val = +val || 0;
+
+    if (val > 0xfffffff || val < 0) throw new Error('Unsupported');
+
+    // realloc
+    let length = this.length || 16;
+    while (length < this.pos + 4) length *= 2;
+    if (length !== this.length) {
+      const buf = new Uint8Array(length);
+      buf.set(this.buf);
+      this.buf = buf;
+      this.length = length;
+    }
+
+    this.buf[this.pos++] = (val & 0x7f) | (val > 0x7f ? 0x80 : 0);
+    if (val <= 0x7f) return;
+    this.buf[this.pos++] = ((val >>>= 7) & 0x7f) | (val > 0x7f ? 0x80 : 0);
+    if (val <= 0x7f) return;
+    this.buf[this.pos++] = ((val >>>= 7) & 0x7f) | (val > 0x7f ? 0x80 : 0);
+    if (val <= 0x7f) return;
+    this.buf[this.pos++] = (val >>> 7) & 0x7f;
+  },
+
+  writeMessage(tag, fn, obj) {
+    this.writeVarint((tag << 3) | 2);
```

Comment:
```suggestion
    this.writeVariant((tag << 3) | 2);
```

(also need to change implementation

Reply:
or is it actually a "varint"?

Reply:
It is `varint` :wink: It's a variable-length integer ( https://developers.google.com/protocol-buffers/docs/encoding#varints )